app/server/controller/src/main/java/io/syndesis/server/controller/integration/BaseIntegrationController.java

```diff
@@ -181,7 +181,7 @@ private void checkIntegrationStatus(IntegrationDeployment integrationDeployment)
         IntegrationDeployment reconciled = reconcileDeployment(integrationDeployment);
         IntegrationDeploymentState desiredState = reconciled.getTargetState();
         IntegrationDeploymentState currentState = reconciled.getCurrentState();
-        if (!currentState.equals(desiredState)) {
+        if (!currentState.equals(IntegrationDeploymentState.Error) && !currentState.equals(desiredState)) {
```

Comment:
So here we're saying that if the current state is not error we'll reconcile by invoking the handler. @rhuss does that make sense to you? Should we be retrying to reconcile for errors as well?

Reply:
I saw that if we reconcile errors here too the reconcile just overwrites the error and continues with deployment. I also think in case we hit some error we should not try to auto recover from that but wait for another publish request